### ISSUE TITLE
Consider adding opengraph support to markdown frontmatter

### ISSUE THREAD

**Issue by dabukalam** [ceph/ceph.io#295]
It would be useful to be able to specify an opengraph title (and maybe even image link) to markdown frontmatter, to provide control over the title. Right now everything seems to pull from https://github.com/ceph/ceph.io/blob/main/src/src.json#L6:

<img width="712" alt="Screenshot 2021-10-19 at 08 15 52" src="https://user-images.githubusercontent.com/5021691/137861560-cc1c8ec8-0ecd-4091-b19a-81aa9a8eb3e8.png">

**Comment by Thingee** [comment]
Am I missing something to get image preview to work on Twitter? For example this post:

[Source](https://github.com/ceph/ceph.io/tree/main/src/en/news/blog/2022/install-ceph-in-a-raspberrypi-4-cluster)
[Page](https://ceph.io/en/news/blog/2022/install-ceph-in-a-raspberrypi-4-cluster/)

![2022-03-11_11-23](https://user-images.githubusercontent.com/47583/157938733-c2f32abd-16dd-487d-9ed7-329b39b3ed94.png)

**Comment by adamduncan** [comment]
@Thingee Looks like you've done the right thing here by ensuring there's a relative `image` property in the post's frontmatter. This then gets pulled into the `_base` template's [`og:image`](https://github.com/ceph/ceph.io/blob/main/src/_includes/layouts/_base.njk#L91) and [`twitter:image`](https://github.com/ceph/ceph.io/blob/main/src/_includes/layouts/_base.njk#L98) meta tags to enable this image preview.

@Pete-Robelou The issue here might be relative URLs. The `image` being relative to the blog post for purposes of the hero image is sound, but I believe we want the meta tag image URLs to be absolute. Do we want to be preprending the current page's URL to `image` if it exists, and `site.url` to the fallback if not? E.g. In this ballpark?

```html
<meta property="og:image" content="{% if image %}{{ site.url }}{{ page.url }}{{ image }}{% else %}{{ site.url }}/assets/favicons/logo-meta-share.png{% endif %}"/>
...
<meta property="twitter:image" content="{% if image %}{{ site.url }}{{ page.url }}{{ image }}{% else %}{{ site.url }}/assets/favicons/logo-meta-share.png{% endif %}"/>
```

**Comment by Pete-Robelou** [comment]
> Do we want to be preprending the current page's URL to image if it exists, and site.url to the fallback if not?

@adamduncan Yeah, you're right. If we make the meta tag image URLs absolute that should resolve the issue.

Your code snippet should work and just needs to replace the relevant lines in `/src/_includes/layouts/_base.njk`.

**Comment by anthonyeleven** [comment]
@bluikko  thoughts?

**Comment by bluikko** [comment]
The later discussion about the image issue may likely have been fixed by #528 but I have no Xitter account to verify.  
It does change the image URL from relative to absolute as suggested in https://github.com/ceph/ceph.io/issues/295#issuecomment-1067801479, but does hardcode the site name unlike suggestion in https://github.com/ceph/ceph.io/issues/295#issuecomment-1066930008.

Unless there is a bug somewhere I believe the other points in the OP issue are procedural and require the blog writer to include the correct metadata in the document.

**Comment by bluikko** [comment]
The [suggestion](https://github.com/ceph/ceph.io/issues/295#issuecomment-946547751) to document how writers could add frontmatter (metadata) is not addressed:

> can you please confirm? Perhaps something to document in the README?

I can open a new issue specifically for that addition as this one has a wide-ranging discussion. Currently the main README [mentions](https://github.com/bluikko/ceph.io/tree/main#page-data) frontmatter only briefly.  
Then this PR could be closed.
